src/java.desktop/share/classes/javax/swing/JComponent.java

```diff
@@ -657,6 +657,7 @@ public JComponent() {
 
     /**
      * Resets the UI property to a value from the current look and feel.
+     * Since the default implementation of this method doesn't do anything,
```

Comment:
The reason why this method should be overridden is that the subclasses only can implement it properly, and it is not because the current method is empty.

Probably we can move the new text to the @implnote tag?

Reply:
Yet specifying that this method is empty implies overridden methods do not need to call `super.updateUI()`.

Reply:
But probably it might be changed in the future? If for example, we will move some common functionality from the standard subclasses to this method.

Reply:
I hasn't changed over all these years… In addition to that, the specification says you *must* override and shows the example without calling `super.updateUI()`.
My take is it's unlikely to change. Yet I understand your concern: Stated explicitly `JComponent.updateUI` is empty, there's no way back. So being cautious wouldn't hurt. From this point of view, `@implnote` still gives some flexibility.

Reply:
`@implnote` added

Reply:
It feels wrong with this `@implNote`.

Subclasses *must still override* the default implementation. Should this statement be included in the regular spec text as it was before? Implementation note could say, “The default implementation doesn't do anything.”

Or alternatively, this change can be rejected at all, leaving the spec as it is now.

Reply:
> It feels wrong with this `@implNote`.
> 
> Subclasses _must still override_ the default implementation. Should this statement be included in the regular spec text as it was before? Implementation note could say, “The default implementation doesn't do anything.”
> 
> Or alternatively, this change can be rejected at all, leaving the spec as it is now.

To make it clearer, what I suggest is keeping the old text as it was and adding `@implNote` below like this:
```
    /**
     * Resets the UI property to a value from the current look and feel.
     * <code>JComponent</code> subclasses must override this method
     * like this:
     * <pre>
     *   public void updateUI() {
     *      setUI((SliderUI)UIManager.getUI(this);
     *   }
     *  </pre>
     * 
     * @implNote The default implementation of this method doesn't do anything.
     *
     * @see #setUI
     * @see UIManager#getLookAndFeel
     * @see UIManager#getUI
     */
```

This way, it's more *highlighted* that `JComponent` subclasses must override `updateUI` as shown for the correct behaviour. Then the implementation note will add the missing detail: The default implementation does nothing. So it's not necessary to call the super method from the overridden one.

What do you, @prsadhuk and @mrserb, think?

Reply:
OK. Seems fair enough for me.